### ISSUE TITLE
GitHub API limits causes Reasoner Validator to occasionally fail upon import

### ISSUE THREAD

**Issue by CaseyTa** [NCATSTranslator/reasoner-validator#94]
**Describe the bug**
On a few occasions over the last few months, I've had Reasoner Validator fail to import for me due to GitHub API [rate limits](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting) with the following exception (this is from reasoner validator 3.8.1):

```
cohd/trapi/reasoner_validator_ext.py:10: in <module>
    from reasoner_validator.validator import TRAPIResponseValidator
/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/reasoner_validator/__init__.py:1: in <module>
    from reasoner_validator.versioning import SemVer
/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/reasoner_validator/versioning.py:19: in <module>
    versions = [release["tag_name"] for release in release_data]
/opt/hostedtoolcache/Python/3.11.4/x64/lib/python3.11/site-packages/reasoner_validator/versioning.py:19: in <listcomp>
    versions = [release["tag_name"] for release in release_data]
E   TypeError: string indices must be integers, not 'str'
```

This exception occurs from versioning.py since the response to 
```response = requests.get(f"https://api.github.com/repos/{GIT_ORG}/{GIT_REPO}/releases")``` 
has json data:
```
{'message': "API rate limit exceeded for 45.152.182.233. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)", 'documentati
on_url': 'https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting'}
```

The first time I noticed this issue (a few months ago) was when I had a bug in my own code that caused COHD to constantly crash and re-deploy, and hence repeatedly imported reasoner validator. This morning I saw this error from my continuous monitoring tests that were run from Github Actions, which only ran once. This job re-ran successfully at a later time.

This bug is not of immediate concern to me, but you may want to check with ITRB / Architecture if there may be some circumstances where Reasoner Validator is frequently imported within an hour. 

**To Reproduce**
GitHub has a limit of 60 calls/hr from the same IP, so I was able to reproduce with code like
```Python
from importlib import reloadlib
from reasoner_validator import versioning
for i in range(100):
    print(i)
    reload(versioning)
```
Note that this will block your IP from calling Github APIs for an hour, so I did this over a VPN connection so that my normal IP address doesn't get blocked.

**Comment by RichardBruskiewich** [comment]
Hi @CaseyTa (cc: @edeutsch, @putmantime, @sierra-moxon, @cbizon),

Thank you for this bug report. Sorry for your troubles here.

As you note, this is an external Github API access issue.

I'm personally familiar with this issue which is a curious transient Github API call threshold error which goes away after a timeout period (i.e. 1 hour).

That said, I'll see what I can now do to "fix" it with a design change, likely, some local file/repo caching of the versions, which really don't change often.

Hang in there... I'll see what I can do today.

Cheers
Richard

**Comment by edeutsch** [comment]
I forgot about this, but have stumbled into it in the past as well. It occurs to me that ARAX never seems to suffer from this. I wonder if AWS instances are exempted somehow. ARAX should be constantly running into this, but it never has.. I think I only time I've hit it is while testing on my own desktop.

**Comment by RichardBruskiewich** [comment]
Should now be resolved with **reasoner-validator v3.8.3** (@CaseyTa, @edeutsch, @putmantime, @sierra-moxon, @cbizon)
